tfx/components/example_gen/base_example_gen_executor.py

```diff
@@ -33,13 +34,17 @@
 
 
 def _GeneratePartitionKey(record: Union[tf.train.Example,
-                                        tf.train.SequenceExample, bytes],
+                                        tf.train.SequenceExample, bytes,
+                                        pa.Table],
                           split_config: example_gen_pb2.SplitConfig) -> bytes:
   """Generates key for partition."""
 
   if not split_config.HasField('partition_feature_name'):
     if isinstance(record, bytes):
       return record
+    # pyarrow tables are passed as dict to this funtion.
+    if isinstance(record, dict):
```

Comment:
s/dict/pa.Table?

Reply:
so looking a bit closer to beam, it seems I was wrong, the `ReadFromParquet` returns a Pcollection of dictionaries. The `ReadFromParquetBatched` returns a PColleciton of `pa.Table`. Unfortnately there's no Sink to write pa.Table, `beam.io.WriteToParquet` expects dictionaries. 

I am gonna change the `pa.Table` types to dict